diff --git a/node_modules/@react-router/dev/dist/cli/index.js b/node_modules/@react-router/dev/dist/cli/index.js
index 9516410..4dd11cb 100644
--- a/node_modules/@react-router/dev/dist/cli/index.js
+++ b/node_modules/@react-router/dev/dist/cli/index.js
@@ -864,6 +864,7 @@ var init_route = __esm({
 });
 
 // typegen/generate.ts
+const getTypescriptSafePath = (path8) => path8.replace(/\.(js|ts)x?$/, ".js");
 function generate2(ctx, route) {
   const lineage2 = lineage(ctx.config.routes, route);
   const fullpath2 = fullpath(lineage2);
@@ -875,9 +876,9 @@ function generate2(ctx, route) {
       getTypesPath(ctx, parent)
     );
     const indent = i === 0 ? "" : "  ".repeat(2);
-    let source = noExtension(rel);
+    let source = getTypescriptSafePath(rel);
     if (!source.startsWith("../")) source = "./" + source;
-    return `${indent}import type { Info as Parent${i} } from "${source}.js"`;
+    return `${indent}import type { Info as Parent${i} } from "${source}"`;
   }).join("\n");
   return import_dedent.default`
     // React Router generated types for route:
@@ -887,7 +888,7 @@ function generate2(ctx, route) {
 
     ${parentTypeImports}
 
-    type Module = typeof import("../${Pathe2.filename(route.file)}.js")
+    type Module = typeof import("../${getTypescriptSafePath(Path4.basename(route.file))}")
 
     export type Info = {
       parents: [${parents.map((_, i) => `Parent${i}`).join(", ")}],
@@ -943,7 +944,6 @@ var init_generate = __esm({
     init_paths();
     init_params();
     init_route();
-    noExtension = (path8) => Path4.join(Path4.dirname(path8), Pathe2.filename(path8));
   }
 });
 
